### DISCUSSION TITLE
Anyone had success using the python asyncspeak APIs to generate speech?

### DISCUSSION THREAD

**Discussion by brentlyjr** [deepgram/community#681] [General help]
Looking to generate speech asynchronously in python and get the result as a stream. All I can see in the documentation is examples to file:

        response = await deepgram.asyncspeak.v("1").save(
            filename, SPEAK_OPTIONS, options
        )

Tried to walk some of the Deepgram python library files but don't see anything helpful that would indicate this is supported. Anyone have any tips or pointers if this is possible. Thank you!

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by brentlyjr** [comment]
I got something working directly posting to the URL, but not sure this is supported in the native Python SDK.

```
import requests
import time

# Mulaw, 8k, mono encoding
DEEPGRAM_URL = "https://api.deepgram.com/v1/speak?model=aura-asteria-en&encoding=mulaw&container=none&sample_rate=8000"
DEEPGRAM_API_KEY = "DEEPGRAM_API_KEY"

payload = {
    "text": "Hello, how can I help you today? My name is Emily and I'm very glad to meet you. What do you think of this new text-to-speech API?"
}

headers = {
    "Authorization": f"Token {DEEPGRAM_API_KEY}",
    "Content-Type": "application/json"
}

audio_file_path = "deepgram.bin"  # Path to save the audio file

start_time = time.time()

with open(audio_file_path, 'wb') as file_stream:
    response = requests.post(DEEPGRAM_URL, headers=headers, json=payload, stream=True)
    for chunk in response.iter_content(chunk_size=1024):
        if chunk:
            print("Time to First Byte: ", time.time() - start_time)
            file_stream.write(chunk) # Write each chunk of audio data to the file

print("Audio download complete")

```
